test/e2e/suite_test.go

```diff
@@ -1109,6 +1159,142 @@ func addDcToCluster(t *testing.T, ctx context.Context, namespace string, f *fram
+			Name:      DcPrefix(t, f, dc1Key) + "-stargate",
+		},
+	}
+	checkStargateReady(t, f, ctx, sg1Key)
+
+	reaper1Key := framework.ClusterKey{
+		K8sContext: f.DataPlaneContexts[0],
+		NamespacedName: types.NamespacedName{
+			Namespace: namespace,
+			Name:      DcPrefix(t, f, dc1Key) + "-reaper",
+		},
+	}
+	checkReaperReady(t, f, ctx, reaper1Key)
+
+	dcSize := 2
+	t.Log("create keyspaces")
+	_, err = f.ExecuteCql(ctx, f.DataPlaneContexts[0], namespace, kc.SanitizedName(), DcPrefix(t, f, dc1Key)+"-default-sts-0",
+		fmt.Sprintf("CREATE KEYSPACE ks1 WITH REPLICATION = {'class' : 'NetworkTopologyStrategy', 'dc1' : %d}", dcSize))
+	require.NoError(err, "failed to create keyspace")
+
+	_, err = f.ExecuteCql(ctx, f.DataPlaneContexts[0], namespace, kc.SanitizedName(), DcPrefix(t, f, dc1Key)+"-default-sts-0",
+		fmt.Sprintf("CREATE KEYSPACE ks2 WITH REPLICATION = {'class' : 'NetworkTopologyStrategy', 'dc1' : %d}", dcSize))
+	require.NoError(err, "failed to create keyspace")
+
+	t.Log("add dc2 to cluster")
+
+	require.Eventually(func() bool {
+		kc := &api.K8ssandraCluster{}
+		err = f.Client.Get(ctx, kcKey, kc)
+		if err != nil {
+			t.Logf("failed to add DC: failed to get K8ssandraCluster: %v", err)
+			return false
+		}
+
+		kc.Spec.Cassandra.Datacenters = append(kc.Spec.Cassandra.Datacenters, api.CassandraDatacenterTemplate{
+			Meta: api.EmbeddedObjectMeta{
+				Name: "dc2",
+			},
+			K8sContext: f.DataPlaneContexts[1],
+			Size:       int32(dcSize),
+		})
+		// annotations.AddAnnotation(kc, api.DcReplicationAnnotation, fmt.Sprintf("{\"dc2\": {\"ks1\": %d, \"ks2\": %d}}", dcSize, dcSize))
+
+		err = f.Client.Update(ctx, kc)
+		if err != nil {
+			t.Logf("failed to add DC: %v", err)
+			return false
+		}
+
+		return true
+	}, 30*time.Second, 1*time.Second, "timed out waiting to add DC to K8ssandraCluster")
+
+	dc2Key := framework.ClusterKey{K8sContext: f.DataPlaneContexts[1], NamespacedName: types.NamespacedName{Namespace: namespace, Name: "dc2"}}
+	checkDatacenterReady(t, ctx, dc2Key, f)
+
+	t.Log("retrieve database credentials")
+	username, password, err := f.RetrieveDatabaseCredentials(ctx, f.DataPlaneContexts[0], namespace, kc.SanitizedName())
+	require.NoError(err, "failed to retrieve database credentials")
+
+	t.Log("check that nodes in dc1 see nodes in dc2")
+	pod := DcPrefix(t, f, dc1Key) + "-default-sts-0"
+	count := dcSize * 2
+	checkNodeToolStatus(t, f, f.DataPlaneContexts[0], namespace, pod, count, 0, "-u", username, "-pw", password)
+
+	assert.NoError(err, "timed out waiting for nodetool status check against "+pod)
+
+	t.Log("check nodes in dc2 see nodes in dc1")
+	pod = DcPrefix(t, f, dc2Key) + "-default-sts-0"
+	checkNodeToolStatus(t, f, f.DataPlaneContexts[1], namespace, pod, count, 0, "-u", username, "-pw", password)
+
+	assert.NoError(err, "timed out waiting for nodetool status check against "+pod)
+
+	keyspaces := []string{"system_auth", stargate.AuthKeyspace, "reaper_db"}
+	for _, ks := range keyspaces {
```

Comment:
You can check that the expected CassandraTask is successfully executed. It would be good to also add/update integration tests.

Reply:
yes, definitely worth checking the CassandraTask ran successfully.